statistics/most_distinct_dates_competed_on.rb

```diff
@@ -0,0 +1,42 @@
+require_relative "../core/statistic"
+
+class MostDistinctDatesCompetedOn < Statistic
+  def initialize
+    @title = "Most distinct dates competed on"
+    @table_header = { "Dates" => :right, "Person" => :left}
+  end
+
+  def query
+    <<-SQL
+      SELECT
+        attended_dates,
+        CONCAT('[', person.name, '](https://www.worldcubeassociation.org/persons/', person.id, ')') person_link
+      FROM (
+        SELECT
+          COUNT(DISTINCT DATE_FORMAT(competition_date, '%m%d')) AS attended_dates,
+          personId
+        FROM (
+          SELECT Results.personId, competition_dates.competition_date
+          FROM Results
+          JOIN (
+            SELECT
+              Competitions.id AS competitionId,
+              DATE_ADD(Competitions.start_date, INTERVAL nums.num DAY) AS competition_date
+            FROM Competitions
+            JOIN (
+              SELECT (a.num + b.num * 10 + c.num * 100) AS num
+              FROM (SELECT 0 AS num UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) AS a
+              CROSS JOIN (SELECT 0 AS num UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) AS b
+              CROSS JOIN (SELECT 0 AS num UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) AS c
```

Comment:
It is a fair assumption that a competition takes less than 10 days :)

```suggestion
              SELECT 0 AS num UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9
```

Reply:
Hahaha, yes indeed! I guess AIs can be pretty dumb when they're being smart.

Fixed it, and it still works the same!